include/rmm/device_scalar.hpp

```diff
@@ -60,28 +56,86 @@ class device_scalar {
    * @return T The value of the scalar after synchronizing its stream
    *---------------------------------------------------------------------------**/
   T value() const {
-    T host_value{};
-    auto status = cudaMemcpyAsync(&host_value, buff.data(), sizeof(T),
-                             cudaMemcpyDefault, buff.stream());
-    if (cudaSuccess != status) {
-      throw std::runtime_error{"Device memcpy failed."};
-    }
-    status = cudaStreamSynchronize(buff.stream());
-    if (cudaSuccess != status) {
-      throw std::runtime_error{"Stream sync failed."};
-    }
-    return host_value;
+    return _value<true>(buff.stream());
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Copies the value from device to host and returns the value.
+   *
+   * @return T The value of the scalar after synchronizing its stream
+   * @param stream CUDA stream on which to perform the copy
+   *---------------------------------------------------------------------------**/
+  T value(cudaStream_t stream) const {
+    return _value<true>(stream);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Copies the value from device to host and returns the value.
+   *
+   * @return T The value of the scalar
+   *---------------------------------------------------------------------------**/
+  T value_async() const {
```

Comment:
Delete this version, since PR #169 now removes the "stored stream" semantics of device_buffer.